frontend/src/pages/Traces/TraceProvider.tsx

```diff
@@ -132,6 +132,19 @@ export const TraceProvider: React.FC<React.PropsWithChildren<Props>> = ({
 		return organizeSpansWithChildren(spans)
 	}, [data?.trace?.trace])
 
+	useEffect(() => {
+		if (spanId) {
+			const selectedSpan = data?.trace?.trace.find(
+				(span) => span.spanID === spanId,
+			)
+
+			if (selectedSpan) {
+				setSelectedSpan(selectedSpan as FlameGraphSpan)
+			}
+		}
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [spanId])
```

Comment:
Unrelated, but noticed that the selected span wasn't updating when clicking different spans on the traces page.